aspnetcore/hosting/ihostingstartup.md

```diff
@@ -0,0 +1,164 @@
+
+```
+<DRIVE>\Users\<USER>\.dotnet\store\x64\netcoreapp2.0\<FEATURE_ASSEMBLY_NAME>\<FEATURE_VERSION>\lib\netcoreapp2.0\
+```
+
+For global use, place the assembly in the .NET Core installation's runtime store:
+
+```
+<DRIVE>\Program Files\dotnet\store\x64\netcoreapp2.0\<FEATURE_ASSEMBLY_NAME>\<FEATURE_VERSION>\lib\netcoreapp2.0\
+```
+
+When deploying the assembly to the runtime store, the symbols file may be deployed as well but isn't required for the feature to work.
+
+**Place the dependencies file**
+
+The implementation's *\*.deps.json* file must be in an accessible location.
+
+For per-user use, place the file in the `additonalDeps` folder of the user profile's `.dotnet` settings: 
+
+```
+<DRIVE>\Users\<USER>\.dotnet\x64\additionalDeps\<FEATURE_ASSEMBLY_NAME>\shared\Microsoft.NETCore.App\2.0.0\
+```
+
+For global use, place the file in the `additonalDeps` folder of the .NET Core installation:
+
+```
+<DRIVE>\Program Files\dotnet\additionalDeps\<FEATURE_ASSEMBLY_NAME>\shared\Microsoft.NETCore.App\2.0.0\
+```
+
+Note the version, `2.0.0`, reflects the version of the shared runtime that the target app uses. The shared runtime is shown in the *\*.runtimeconfig.json* file. In the sample app, the shared runtime is specified in the *HostingStartupSample.runtimeconfig.json* file.
+
+**Set environment variables**
+
+Set the following environment variables in the context of the app that uses the feature.
+
+ASPNETCORE\_HOSTINGSTARTUPASSEMBLIES
+
+Only hosting startup assemblies are scanned for the `HostingStartupAttribute`. The assembly name of the implementation is provided in this environment variable. The sample app sets this value to `StartupDiagnostics`.
+
+The value can also be set using the [Hosting Startup Assemblies](xref:fundamentals/hosting#hosting-startup-assemblies) host configuration setting.
+
+DOTNET\_ADDITIONAL\_DEPS
+
+The location of the implementation's *\*.deps.json* file.
+
+If the file is placed in the user profile's *.dotnet* folder for per-user use:
+
+```
+<DRIVE>\Users\<USER>\.dotnet\x64\additionalDeps\
+```
+
+If the file is placed in the .NET Core installation for global use, provide the full path to the file:
+
+```
+<DRIVE>\Program Files\dotnet\additionalDeps\<FEATURE_ASSEMBLY_NAME>\shared\Microsoft.NETCore.App\2.0.0\<FEATURE_ASSEMBLY_NAME>.deps.json
+```
+
+The sample app sets this value to:
+
+```
+%UserProfile%\.dotnet\x64\additionalDeps\StartupDiagnostics\
+```
+
+For examples of how to set environment variables for various operating systems, see [Working with multiple environments](xref:fundamentals/environments).
+
+## Sample app
+
+The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/hosting/ihostingstartup/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample)) uses `IHostingStartup` to create a diagnostics tool. The tool adds two middlewares to the app at startup that provide diagnostic information:
```

Comment:
provides ?

Reply:
The "two middlewares" "provide" (plural verb) 'the feature.' I think we're ok here.

Reply:
I guess the reason that I'm focusing on "two middlewares" and not "the tool" is that the object of "that" feels like it refers to the "two middlewares" more than "the tool." This is the way that I would speak the sentence verbally.

Reply:
I agree. That's why I put ?

Reply:
lol :smile: I don't take anything you suggest lightly!